Sources/Sentry/TracesSampler.m

```diff
@@ -0,0 +1,50 @@
+#import "TracesSampler.h"
+#import "SentryOptions.h"
+#import "SentrySamplingContext.h"
+#import "SentryTransactionContext.h"
+
+@implementation TracesSampler {
+    SentryOptions *_options;
+}
+
+- (instancetype)initWithOptions:(SentryOptions *)options random:(id<Random>)random
+{
+    if (self = [super init]) {
+        _options = options;
+        self.random = random;
+    }
+    return self;
+}
+
+- (instancetype)initWithOptions:(SentryOptions *)options
+{
+    return [self initWithOptions:options random:[[Random alloc] init]];
+}
+
+- (SentrySampleDecision)sample:(SentrySamplingContext *)context
+{
+    if (context.transactionContext.sampled != kSentrySampleDecisionUndecided)
+        return context.transactionContext.sampled;
+
+    if (_options.tracesSampler != nil) {
+        NSNumber *callbackDecision = _options.tracesSampler(context);
```

Comment:
Since the traceSampleRate default value is 0, an invalid value shouldn't be treated as 0?

Reply:
For me, it would make sense to treat it as 0. I just want it to be consistent across the whole SDK. Then let's change it to 0 if it's invalid everywhere. https://github.com/getsentry/sentry-cocoa/issues/986